src/collectors/loggly/test/testloggly.py

```diff
@@ -0,0 +1,214 @@
+            'q=mock: bad&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 400),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: unauthorized&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 401),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: forbidden&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 403),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: gone&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 410),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: serverr&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 500),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: not implemented&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 501),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: throttled&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 503),
+            'https://mocksub.loggly.com/apiv2/search?'
+            'q=mock: timeout&from=-10s&until=now':
+                MockResponse({'rsid': {'id': 1}}, 504),
+            'https://mocksub.loggly.com/apiv2/events?rsid=1':
+                MockResponse({
+                    'events': [{'timestamp': 1443092343,
+                                'event': {'mock': {'field': 123.456}}}],
+                    'total_events': 4}, 200)
+        }
+        try:
+            return recorded_responses[args[1]]
+        except:
+            return MockResponse({}, 404)
+
+    def setUp(self):
+        config = get_collector_config('LogglyCollector',
+                                      self.TEST_CONFIG)
+
+        self.collector = LogglyCollector(config, None)
+
+    def test_import(self):
+        self.assertTrue(LogglyCollector)
+
+    # @run_only_if_requirements_present
+    @patch.object(Collector, 'publish_metric')
+    def test(self, publish_mock):
+        patch_requests_get = patch('requests.get',
+                                   side_effect=self.mocked_requests_get)
+        patch_requests_get.start()
+        self.collector.collect()
+        patch_requests_get.stop()
+
+        published_metrics = {
+            'testmetric.mock.field': (123.456, None),
+            'countmetric.mock.field': (123.456, None),
+            'countmetric.count': 4
+        }
+        unpublished_metrics = {
+            'badmetric.mock.field': 0,
+            'unauthorizedmetric.mock.field': 0,
+            'forbiddenmetric.mock.field': 0,
+            'gonemetric.mock.field': 0,
+            'serverrmetric.mock.field': 0,
+            'notimplementedmetric.mock.field': 0,
+            'throttledmetric.mock.field': 0,
+            'timeoutmetric.mock.field': 0,
+        }
+
+        for key, value in published_metrics.iteritems():
```

Comment:
Interesting, so it seems that `testmountstats.py` simply calls `assertPublishedMany` followed by `assertUnpublishedMany`. Are you suggesting that that test is incorrect since it resets the mock in between?

If so, maybe its time we clean up these assertions and have a unified function called `assertMetrics(self, mock, published_dict, unpublished_list=None)`.